src/test/kotlin/org/blueliner/springbootwithkotlin/SpringBootWithKotlinApplicationTests.kt

```diff
@@ -1,13 +1,65 @@
 package org.blueliner.springbootwithkotlin
 
+import org.blueliner.springbootwithkotlin.dto.JokeDto
+import org.blueliner.springbootwithkotlin.service.JokeService
 import org.junit.jupiter.api.Test
+import org.springframework.ai.chat.client.ChatClient
+import org.springframework.ai.chat.client.ChatClient.PromptUserSpec
+import org.springframework.ai.openai.OpenAiChatModel
+import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
 
 @SpringBootTest
 class SpringBootWithKotlinApplicationTests {
 
-	@Test
-	fun contextLoads() {
-	}
+    @Autowired
+    private lateinit var jokeService: JokeService
+
+    @Autowired
+    private lateinit var chat: OpenAiChatModel
+
+    @Test
+    fun testIsJokeReturnsTrueForEachJokeInList() {
+        val jokes = jokeService.getJokesUsingCustomOutput()
+        jokes.forEach { joke ->
+            run {
+                val isJoke = isJoke(joke)
+                println("Joke: ${joke.jokeDescription} ? $isJoke")
+                assertTrue { isJoke }
+            }
+        }
+    }
+
+    @Test
+    fun testIsJokeReturnsFalseForRandomString() {
+        assertFalse { isJoke(JokeDto("Hello my name is Jack")) }
+    }
+
+    @Test
+    fun testIsJokesReturnsTrueForListOfJokes() {
+        val jokes = jokeService.getJokesUsingCustomOutput()
+        assertTrue { isJokes(jokes) }
+    }
+
+
+    private fun isJokes(jokes: List<JokeDto>): Boolean {
+        return ChatClient.create(chat).prompt()
+            .user { u: PromptUserSpec ->
+                u.text("Are these answers jokes {jokes} ?").param("jokes", jokes)
+            }
+            .call()
+            .entity(Boolean::class.java)
+    }
+
```

Comment:
does it use structured outputs under the hood? is it a 100% reliable method?

Reply:
I checked implementation under the hood and yes its use structered outputs
![image](https://github.com/user-attachments/assets/a6c67637-38d1-4716-8696-f5e0f1e77402)